app/assets/javascripts/client/consensus/ConsensusService.js

```diff
@@ -0,0 +1,153 @@
+(function() {
+  'use strict';
+  angular.module('PDRClient')
+      .service('ConsensusService', ConsensusService);
+
+  ConsensusService.$inject = [
+    '$stateParams',
+    '$location',
+    '$q',
+    'Consensus',
+    'AnalysisConsensus',
+    'AnalysisResponse',
+    'Score',
+    'AnalysisConsensusScore',
+    'ProductEntry',
+    'DataEntry'
+  ];
+
+  function ConsensusService($stateParams, $location, $q, Consensus, AnalysisConsensus, AnalysisResponse, Score, AnalysisConsensusScore, ProductEntry, DataEntry) {
+    var service = this;
+
+    service.setContext = function(context) {
+      service.context = context;
+      return service;
+    };
+
+    service.extractId = function() {
+      return $stateParams.assessment_id || $stateParams.analysis_id;
+    };
+
+    service.redirectToIndex = function () {
+      if (service.context === "assessment") {
+        $location.path("/assessments");
+      } else if (service.context === "analysis") {
+        $location.path("/inventories/" + $stateParams.inventory_id + "/analyses");
+      }
+    };
+
+    service.redirectToReport = function () {
+      if (service.context === "assessment") {
+        $location.path("/assessments/" + service.extractId() + "/report");
+      } else if (service.context === "analysis") {
+        $location.path("/inventories/" + $stateParams.inventory_id + "/analyses/" + service.extractId() + "/report");
+      }
+    };
+
+    service.redirectToCreatedConsensus = function (consensusId) {
+      if (service.context === "assessment") {
+        $location.path("/assessments/" + service.extractId() + "/consensus/" + consensusId);
+      } else if (service.context === "analysis") {
+        $location.path("/inventories/" + $stateParams.inventory_id + "/analyses/" + service.extractId() + "/consensus/" + consensusId);
+      }
+    };
+
+    service.createConsensus = function () {
+      if (service.context === "assessment") {
+        return Consensus.create({
+          assessment_id: service.extractId()
+        }, {}).$promise;
+      } else if (service.context === "analysis") {
+        return AnalysisResponse.save({
+          inventory_id: $stateParams.inventory_id,
+          analysis_id: service.extractId()
+        }, {}).$promise
+        .then(function () {
+          return AnalysisConsensus.create({
+            inventory_id: $stateParams.inventory_id,
+            analysis_id: service.extractId()
+          }, {}).$promise;
+        });
+      }
```

Comment:
I agree with you but am a lot less concerned. Should you need to wire in a consensus for a new type of tool, you're bound to come across `ConsensusService` sooner rather than later by virtue of needing to inject it as a dependency, so these context-depending operations will become obvious sooner rather than later.
That said, we could raise an exception on every method call until `service.context` is no longer undefined — this guarantees the user is aware of the service's needs. @bithavoc, @MakotoTheKnight, is this, as well as other potential improvements to these context-dependent services, something we want to focus on now?